src/main/java/de/tum/in/www1/artemis/service/connectors/LtiService.java

```diff
@@ -95,6 +97,19 @@ public LtiService(UserCreationService userCreationService, UserRepository userRe
         this.artemisAuthenticationProvider = artemisAuthenticationProvider;
         this.ltiUserIdRepository = ltiUserIdRepository;
         this.response = response;
+        this.client = HttpClientBuilder.create().build();
+    }
+
+    public LtiService(UserCreationService userCreationService, UserRepository userRepository, LtiOutcomeUrlRepository ltiOutcomeUrlRepository, ResultRepository resultRepository,
```

Comment:
why do we need a second constructor? I think we should avoid this in services

Reply:
we need that to mock the httpclient, otherwise we cannot cover the test lines 455-458

Reply:
I agree that it might be necessary to inject HttpClient, but this does not answer that a second constructor is needed. Please remove the 2nd constructor again

Reply:
https://cloudacademy.com/course/java-test-driven-development-td-using-junit-1251/mockito-introduced-to-mock-apache-httpclient-http-calls/

https://stackoverflow.com/questions/20542361/mocking-apache-httpclient-using-mockito

Reply:
I mean thats just how its commonly done, we can also skip these changes and stay with the 88% covered

Reply:
I would still like to avoid it, I consider it as bad practice in our project.

Reply:
removed